nexus/db-queries/src/db/datastore/region_snapshot_replacement.rs

```diff
@@ -894,9 +894,7 @@ impl DataStore {
                         .check_if_exists::<RegionSnapshotReplacementStep>(
                             region_snapshot_replacement_step_id,
                         )
-                        .execute_and_check(
```

Comment:
:grimacing: yeah this isn't great, it runs this command on a separate connection from the one used for the transaction itself